marlowe/test/Spec/Marlowe/Serialization/ExtendedJson.hs

```diff
@@ -0,0 +1,50 @@
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-- | This suite tests the Json serialization of the Marlowe extended module
+--
+-----------------------------------------------------------------------------
+{-# LANGUAGE OverloadedStrings #-}
+
+module Spec.Marlowe.Serialization.ExtendedJson
+  ( tests
+  ) where
+
+import Data.Aeson (eitherDecodeFileStrict)
+import Language.Marlowe.Extended.V1 (Contract, Module)
+import System.Directory (getCurrentDirectory)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (assertFailure, testCase)
+
+tests :: TestTree
+tests = testGroup "Extended Contract Serialization"
+    [ testCase "Golden Swap contract" testGoldenSwapContract
```

Comment:
Instead of hard-coding each golden test, could we have the test code discover all of the JSON files in the golden directory and then test each of them?

Reply:
I've added a comment to use a `Proxy` if we find ourselves doing more than these 2 tests. But I'm not sure I'm able just to do a smart discovery because we need to provide the type that we are trying to decode for each file.